### PR TITLE
Add dynamic aspects of configuring Knative Serving

### PR DESCRIPTION
**Pull request by markusthoemmes**
This adds the more dynamic aspects of configuring Knative Serving in that it needs clients to check the cluster's actual state.

This has two commits:
1. All the generation shenanigans.
2. The actual changes to the Knative Serving extension.

### CODE REVIEW

File: openshift-knative-operator/pkg/serving/extension.go [hunk 1]
```diff
@@ -25,6 +36,23 @@ func (e *extension) Transformers(v1alpha1.KComponent) []mf.Transformer {
...
+	if domain != "" {
+		common.Configure(&ks.Spec.CommonSpec, "domain", domain, "")
+	}
+
+	// Attempt to locate kibana route which is available if openshift-logging has been configured
+	route, err := e.ocpclient.RouteV1().Routes("openshift-logging").Get("kibana", metav1.GetOptions{})
+	if err == nil && len(route.Status.Ingress) > 0 && route.Status.Ingress[0].Host != "" {
+		common.Configure(&ks.Spec.CommonSpec, "observability", "logging.revision-url-template",
+			fmt.Sprintf(loggingURLTemplate, route.Status.Ingress[0].Host))
+	}
```
Comment (matzew) [inline openshift-knative-operator/pkg/serving/extension.go:54]: I think this looks more compatc that on the `serving.go` file on the `knative-operator`.

File: openshift-knative-operator/pkg/serving/extension.go [hunk 2]
```diff
@@ -25,6 +36,23 @@ func (e *extension) Transformers(v1alpha1.KComponent) []mf.Transformer {
...
 
+	// Fetch the proper domain.
+	ingress, err := e.ocpclient.ConfigV1().Ingresses().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch cluster config: %w", err)
+	}
+	domain := ingress.Spec.Domain
+	if domain != "" {
+		common.Configure(&ks.Spec.CommonSpec, "domain", domain, "")
+	}
```
Comment (matzew) [inline openshift-knative-operator/pkg/serving/extension.go:47]: wanna stash that to a func?

Comment (markusthoemmes) [reply openshift-knative-operator/pkg/serving/extension.go]: Done

File: openshift-knative-operator/pkg/serving/extension.go (hunk 1 above)
Comment (markusthoemmes) [reply openshift-knative-operator/pkg/serving/extension.go]: Refactored into a function for clarity.

**Comment by markusthoemmes** [comment]
/retest

Known issues over night.

**Comment by markusthoemmes** [comment]
/retest

Failed to create namespaces. That one is REALLY annoying :thinking:

**Pull request merged**